library-app/src/components/Layout/Modal.tsx

```diff
@@ -0,0 +1,28 @@
+import ReactDOM from 'react-dom'
+import styles from './Modal.module.css'
+
+const portalDiv = document.getElementById('portal') as HTMLElement
+
+export interface ModalProps {
+  open: boolean
+  children: React.ReactNode
+  onClose: React.MouseEventHandler
+}
+
+const Modal = ({ open, children, onClose }: ModalProps) => {
+  if (!open) return null
```

Comment:
Null at function that should return tsx code? You can handle this open / closed state at the parent component and show / hide modal there